src/qibo/backends/__init__.py

```diff
@@ -0,0 +1,113 @@
+from qibo import config
+from qibo.config import raise_error, log, warnings, BACKEND_NAME
+from qibo.backends.numpy import NumpyBackend
+from qibo.backends.tensorflow import TensorflowBackend
+
+
+CONSTRUCTED_BACKENDS = {}
+def construct_backend(name):
+    if name not in CONSTRUCTED_BACKENDS:
+        if name == "numpy":
+            CONSTRUCTED_BACKENDS["numpy"] = NumpyBackend()
+        elif name == "tensorflow":
+            CONSTRUCTED_BACKENDS["tensorflow"] = TensorflowBackend()
+        else:
+            raise_error(ValueError, "Unknown backend name {}.".format(name))
+    return CONSTRUCTED_BACKENDS.get(name)
+
+numpy_backend = construct_backend("numpy")
+numpy_matrices = numpy_backend.matrices
+
+if BACKEND_NAME == "tensorflow":
+    K = TensorflowBackend()
+else: # pragma: no cover
+    # CI uses tensorflow as default backend
+    K = NumpyBackend()
+
+def set_backend(backend="custom"):
+    """Sets backend used to implement gates.
+
+    Args:
+        backend (str): possible options are 'custom' for the gates that use
```

Comment:
Here I would be more explicit with the current supported options for each backend (numpy, tensorflow).

Reply:
I added all currently supported backends in this string.